plugins/catalog-react/src/components/EntityRefLink/humanize.ts

```diff
@@ -40,13 +40,22 @@ export function humanizeEntityRef(
   if ('metadata' in entityRef) {
     kind = entityRef.kind;
     namespace = entityRef.metadata.namespace;
-    name = entityRef.metadata.name;
+    if (
+      entityRef.spec &&
+      'profile' in entityRef.spec &&
+      entityRef.spec.profile instanceof Object &&
+      'displayName' in entityRef.spec.profile &&
+      entityRef.spec.profile.displayName
+    ) {
+      name = entityRef.spec.profile.displayName;
+    } else {
+      name = entityRef.metadata.title ?? entityRef.metadata.name;
+    }
```

Comment:
OK we talked among ourselves a bit. We feel that the `humanizeEntityRef` function, both by implication of its name and by its current usages, is specifically for refs and these title/displayName concerns should probably not be mixed into it at all.

There's also the fact that for a lot of the current use cases, such as [in the `EntityOwnerPicker`](https://github.com/backstage/backstage/blob/905141aa08c9c9aba8cea4d4fc4c1cdbc5eb1aa7/plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx#L85) we only have access to the ref, not the entity, so the possibility to leverage something like the suggested code would have to be accompanied with code changes elsewhere to fetch the actual entities which is a bigger thing.

What we could envision then is,

- Leave `humanizeEntityRef` as it was
- Add a `formatLabel?: (entity: Entity) => string` or similar prop to each of the picker components in question so that people can choose the way that they want to render things
- The point above may also require some code changes elsewhere in those pickers

In the future, we can think about if there's some way to supply a single formatter centrally, that several components use. But for now that is probably a step too far to invent.

Reply:
Hi, and thanks for your attention to this. I wasn't sure if it would sell as-is, but at least we've now opened up the conversation. So would we handle this as an issue for which I can rework this PR if someone else doesn't get to it sooner? Or should we try to go ahead and push through the simplest thing that will work, leaving the potential task of fetching actual entities to a given consumer-supplied `formatLabel` implementation?